src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -218,6 +227,14 @@ public CodeWriter emit(CodeBlock codeBlock) throws IOException {
 
         case "$T":
           TypeName typeName = (TypeName) codeBlock.args.get(a++);
+          if (typeName instanceof ClassName) {
+            ClassName candidate = (ClassName) typeName;
+            if (staticImportClassNames.contains(candidate.canonicalName)) {
+              checkState(potentiallyStaticImported == null, "pending type for static import?!");
```

Comment:
I'll fix/guard that by peeking into the next part and check whether `!part.startsWith("$")` -- ensuring that only if a `$T` is followed by a default case, the emittion of `$T` is defered.